syntax = "proto3";

import "google/protobuf/any.proto";

package ficus;

message GrpcPipeline {
    repeated GrpcPipelinePartBase parts = 1;
}

message GrpcPipelinePartBase {
    oneof part {
        GrpcPipelinePart defaultPart = 1;
        GrpcParallelPipelinePart parallelPart = 2;
    }
}

message GrpcPipelinePart {
    string name = 1;
    GrpcPipelinePartConfiguration configuration = 2;
}

message GrpcPipelinePartConfiguration {
    repeated GrpcConfigurationValue configurationParameters = 1;
}

message GrpcConfigurationValue {
    string name = 1;
    google.protobuf.Any value = 2;
}

message GrpcParallelPipelinePart {
    repeated GrpcPipelinePartBase pipelineParts = 1;
}

message GrpcParallelPipelineParts {
    repeated GrpcParallelPipelinePart pipeline = 1;
}